src/lib/error.ts

```diff
@@ -0,0 +1,33 @@
+/* FluxxChat-palvelin
+ * Copyright (C) 2019 Helsingin yliopisto
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+export default class ErrorMessage extends Error {
+	public internal: boolean;
+	public message: string;
+
+	constructor(opts: ErrorMessageOpts) {
+		super(opts.message);
+		this.internal = opts.internal;
+		this.message = opts.message;
+		this.stack = new Error().stack;
+	}
+}
+
+export interface ErrorMessageOpts {
```

Comment:
Don't export this if it's not used anywhere outside this file. If we want to change the name, for example, we would have to check for references. If it's not exported, we immediately know it's not used anywhere else.